http://gcc.gnu.org/PR38627

make sure objects in libgcc.a have GNU-stack markings:
!WX --- ---  4.3.2/_ashiftrt.o
!WX --- ---  4.3.2/_ashiftrt_n.o
!WX --- ---  4.3.2/_ashiftlt.o
!WX --- ---  4.3.2/_lshiftrt.o
!WX --- ---  4.3.2/_movmem.o
!WX --- ---  4.3.2/_movmem_i4.o
!WX --- ---  4.3.2/_mulsi3.o
!WX --- ---  4.3.2/_sdivsi3.o
!WX --- ---  4.3.2/_sdivsi3_i4.o
!WX --- ---  4.3.2/_udivsi3.o
!WX --- ---  4.3.2/_udivsi3_i4.o
!WX --- ---  4.3.2/_set_fpscr.o
!WX --- ---  4.3.2/_div_table.o
!WX --- ---  4.3.2/_udiv_qrnnd_16.o
!WX --- ---  4.3.2/_ic_invalidate.o
!WX --- ---  4.3.2/_ic_invalidate_array.o
!WX --- ---  4.3.2/linux-atomic.o

--- gcc/config/sh/lib1funcs.asm
+++ gcc/config/sh/lib1funcs.asm
@@ -34,6 +34,11 @@
 !! recoded in assembly by Toshiyasu Morita
 !! tm@netcom.com
 
+#if defined(__ELF__) && defined(__linux__)
+.section .note.GNU-stack,"",%progbits
+.previous
+#endif
+
 /* SH2 optimizations for ___ashrsi3, ___ashlsi3, ___lshrsi3 and
    ELF local label prefixes by J"orn Rennecke
    amylaar@cygnus.com  */
--- gcc/config/sh/linux-atomic.asm
+++ gcc/config/sh/linux-atomic.asm
@@ -138,3 +138,6 @@
 ATOMIC_FETCH_AND_COMBOP(nand,and,not,4,l,mov)
 
 #endif /* ! __SH5__ */
+
+.section .note.GNU-stack,"",%progbits
+.previous
